materialize/resources/resource_connection_postgres.go

```diff
@@ -0,0 +1,395 @@
+	postgresSSLMode        string
+	postgresAWSPrivateLink string
+}
+
+func newConnectionPostgresBuilder(connectionName, schemaName, databaseName string) *ConnectionPostgresBuilder {
+	return &ConnectionPostgresBuilder{
+		connectionName: connectionName,
+		schemaName:     schemaName,
+		databaseName:   databaseName,
+	}
+}
+
+func (b *ConnectionPostgresBuilder) ConnectionName(connectionName string) *ConnectionPostgresBuilder {
+	b.connectionName = connectionName
+	return b
+}
+
+func (b *ConnectionPostgresBuilder) SchemaName(schemaName string) *ConnectionPostgresBuilder {
+	b.schemaName = schemaName
+	return b
+}
+
+func (b *ConnectionPostgresBuilder) ConnectionType(connectionType string) *ConnectionPostgresBuilder {
+	b.connectionType = connectionType
+	return b
+}
+
+func (b *ConnectionPostgresBuilder) PostgresDatabase(postgresDatabase string) *ConnectionPostgresBuilder {
+	b.postgresDatabase = postgresDatabase
+	return b
+}
+
+func (b *ConnectionPostgresBuilder) PostgresHost(postgresHost string) *ConnectionPostgresBuilder {
+	b.postgresHost = postgresHost
+	return b
+}
+
+func (b *ConnectionPostgresBuilder) PostgresPort(postgresPort int) *ConnectionPostgresBuilder {
+	b.postgresPort = postgresPort
+	return b
+}
+
+func (b *ConnectionPostgresBuilder) PostgresUser(postgresUser string) *ConnectionPostgresBuilder {
+	b.postgresUser = postgresUser
+	return b
+}
+
+func (b *ConnectionPostgresBuilder) PostgresPassword(postgresPassword string) *ConnectionPostgresBuilder {
+	b.postgresPassword = postgresPassword
+	return b
+}
+
+func (b *ConnectionPostgresBuilder) PostgresSSHTunnel(postgresSSHTunnel string) *ConnectionPostgresBuilder {
+	b.postgresSSHTunnel = postgresSSHTunnel
+	return b
+}
+
+func (b *ConnectionPostgresBuilder) PostgresSSLCa(postgresSSLCa string) *ConnectionPostgresBuilder {
+	b.postgresSSLCa = postgresSSLCa
+	return b
+}
+
+func (b *ConnectionPostgresBuilder) PostgresSSLCert(postgresSSLCert string) *ConnectionPostgresBuilder {
+	b.postgresSSLCert = postgresSSLCert
+	return b
+}
+
+func (b *ConnectionPostgresBuilder) PostgresSSLKey(postgresSSLKey string) *ConnectionPostgresBuilder {
+	b.postgresSSLKey = postgresSSLKey
+	return b
+}
+
+func (b *ConnectionPostgresBuilder) PostgresSSLMode(postgresSSLMode string) *ConnectionPostgresBuilder {
+	b.postgresSSLMode = postgresSSLMode
+	return b
+}
+
+func (b *ConnectionPostgresBuilder) PostgresAWSPrivateLink(postgresAWSPrivateLink string) *ConnectionPostgresBuilder {
+	b.postgresAWSPrivateLink = postgresAWSPrivateLink
+	return b
+}
+
+func (b *ConnectionPostgresBuilder) Create() string {
+	q := strings.Builder{}
+	q.WriteString(fmt.Sprintf(`CREATE CONNECTION %s.%s.%s TO POSTGRES (`, b.databaseName, b.schemaName, b.connectionName))
+
+	q.WriteString(fmt.Sprintf(`HOST '%s'`, b.postgresHost))
+	q.WriteString(fmt.Sprintf(`, PORT %d`, b.postgresPort))
+	q.WriteString(fmt.Sprintf(`, USER '%s'`, b.postgresUser))
+	q.WriteString(fmt.Sprintf(`, PASSWORD SECRET %s`, b.postgresPassword))
```

Comment:
This may need a conditional, ore should be set as a required parameter. According to the docs, it isn't required but not sure if that is true.

https://materialize.com/docs/sql/create-connection/#postgres-options